app/controllers/finance/api/payment_callbacks/stripe_callbacks_controller.rb

```diff
@@ -21,7 +21,13 @@ class MissingStripeEndpointSecret < StripeCallbackError; end
 
   # Undocumented endpoint used for update callbacks of async-authorized payment transactions (mostly due to SCA regulations)
   def create
-    service = Finance::StripePaymentIntentUpdateService.new(current_account, stripe_event)
+    begin
+      service = Finance::StripePaymentIntentUpdateService.new(current_account, stripe_event)
+    rescue ActiveRecord::RecordNotFound
+      # Returning 204 to acknowledge reception even for payments we don't care about
+      # https://issues.redhat.com/browse/THREESCALE-6851
+      return head(:no_content)
+    end
```

Comment:
We could remove the double `return head(:no_content)` and the variable assignment inside of the block with something like
```ruby
def create
  service = begin
    Finance::StripePaymentIntentUpdateService.new(current_account, stripe_event)
  rescue ActiveRecord::RecordNotFound
    SomeThing
  end

  return head(:no_content) if `service is SomeThing` || service.call
  ...
```
Which doesn't look like the best thing possible...

The odd thing for me is that `Finance::StripePaymentIntentUpdateService` can raise an `ActiveRecord::RecordNotFound` when initialised, instead of a custom error, for example.

Reply:
Aside from this controller, `Finance::StripePaymentIntentUpdateService` is only called [here](https://github.com/3scale/porta/blob/master/lib/developer_portal/app/controllers/developer_portal/admin/account/invoices_controller.rb#L40). Do we need to perform the rescue there as well?
Maybe we should change `Finance::StripePaymentIntentUpdateService` so it raises an error when `call` is executed or even set an `errors` attribute to it, if we get an `ActiveRecord::RecordNotFound` when it's initialised, which would make life easier to work with its exceptions, in the sense it'd be handled by the service itself.

Reply:
I made some changes based on your comment.

About `Finance::StripePaymentIntentUpdateService` begin called from [here](https://github.com/3scale/porta/blob/master/lib/developer_portal/app/controllers/developer_portal/admin/account/invoices_controller.rb#L40), we don't need to add the rescue there. That rescue only makes sense in the webhook b/c we want to return a `20X` status code to Stripe for it to not disable the webhook. But out of the webhook we don't want to mask the exception. Anyway, the `RecordNotFound` exception comes from [/app/services/finance/stripe_payment_intent_update_service.rb#L7](https://github.com/3scale/porta/blob/3017562449d3275840cae6afe338a01c1dcca899/app/services/finance/stripe_payment_intent_update_service.rb#L7), as a result of sending it an non existing `payment_intent`, that is, trying to pay an invoice not in our DB. But when called from [/app/controllers/developer_portal/admin/account/invoices_controller.rb#L40](https://github.com/3scale/porta/blob/master/lib/developer_portal/app/controllers/developer_portal/admin/account/invoices_controller.rb#L40), the line above already takes the proper invoice and fails if it doesn't exist, so it will always instantiate the service with a valid `payment_intent`.